docs/source/conf.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+#
+# complexity documentation build configuration file, created by
+# sphinx-quickstart on Tue Jul  9 22:26:36 2013.
+#
+# This file is execfile()d with the current directory set to its containing dir.
+#
+# Note that not all possible configuration values are present in this
+# autogenerated file.
+#
+# All configuration values have a default; values that are commented out
+# serve to show the default.
+
+import xpublish
+import sys
+import os
+
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+# sys.path.insert(0, os.path.abspath('.'))
+
+cwd = os.getcwd()
+parent = os.path.dirname(cwd)
+sys.path.insert(0, parent)
+
+
+# -- General configuration -----------------------------------------------------
+
+# If your documentation needs a minimal Sphinx version, state it here.
+# needs_sphinx = '1.0'
+
+# Add any Sphinx extension module names here, as strings. They can be extensions
+# coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
+extensions = [
+    "sphinx.ext.autodoc",
+    "sphinx.ext.viewcode",
+    "sphinx.ext.autosummary",
+    "sphinx.ext.doctest",
+    "sphinx.ext.intersphinx",
+    "sphinx.ext.extlinks",
+    "numpydoc",
+    "IPython.sphinxext.ipython_console_highlighting",
+    "IPython.sphinxext.ipython_directive",
+]
+
+extlinks = {
+    "issue": ("https://github.com/jhamman/xpublish/issues/%s", "GH#"),
+    "pr": ("https://github.com/jhamman/xpublish/pull/%s", "GH#"),
+}
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ["_templates"]
+
+# Generate the API documentation when building
+autosummary_generate = True
+numpydoc_show_class_members = False
+
+
+# The suffix of source filenames.
+source_suffix = ".rst"
+
+# The encoding of source files.
+# source_encoding = 'utf-8-sig'
+
+# The master toctree document.
+master_doc = "index"
+
+# General information about the project.
+project = "xpublish"
+copyright = "2019 onwards, Joe Hamman and its Contributors"
+author = "Joe Hamman"
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+version = xpublish.__version__
+# The full version, including alpha/beta/rc tags.
+release = xpublish.__version__
```

Comment:
These lines are going to fail since xpublish does not have `__version__` attribute yet. We need to add the following to `__init__.py`

```python
from pkg_resources import DistributionNotFound, get_distribution
try:
    __version__ = get_distribution(__name__).version
except DistributionNotFound:
    # package is not installed
    pass
```

Reply:
This is fixed. I pushed a few changes to your branch